### ISSUE TITLE
Making one to many requests

### ISSUE THREAD

**Issue by rjwhitmer** [knowndecimal/fulfil#31]
Currently the gem is set up to only make one to one relationship PUT requests. However, Fulfil also has one to many relationships that need to be addressed as well. 

One to one relationship requests are made as such: 
`FulfilClient.post(body: [['id', '=', id]], model: {model_name})`

The URI that is generated with this request is as follows:

`POST https://{subdomain}.fulfil.io/api/v2/model/{model_name}`

However, a one to many relationship PUT request through Fulfil's API needs to be as follows:

`FulfilClient.put(body: [[id], {<one_to_many_attribute>: [["add",[<attribute_ids>]]]}], model: {model_name})`

The URI should resemble this:

`PUT https://{subdomain}.fulfil.io/api/v2/model/{model_name}/write`

Likewise, if you'd like to remove an attribute through a one to many, the request should look as follows:

`FulfilClient.put(body: [[id], {<one_to_many_attribute>: [["remove",[<attribute_ids>]]]}], model: {model_name})`

The URI for both adding and removal should look the same.

**Comment by stefanvermaas** [comment]
What is the current URI that is generated in this case?

**Comment by rjwhitmer** [comment]
The current URI either ends with the `model_name`, `search_read`, or `search_count`. I think the biggest change we need is the ability to use `write` as a suffix. It might be beneficial to allow the passing of the `id` and `body` separately since the structure of the request is a bit different than the rest. For example:

Current request body structure (One to One relationship):

`[['foo', '=', 'bar'], ['id', '=', '123']]`

One to Many request body structure for creating a relationship:

URI
`PUT https://{merchant}.fulfil.io/api/v2/model/party.party/write`

`[[id], {"foo": [["add", [1, 2, 3]]] }]`

To remove a relationship:

`[[id], {"foo": [["remove", [1, 2, 3]]] }]`

**Comment by stefanvermaas** [comment]
Yeah, this would mean some architectural changes. I think that's fair. I'll chew on it for a bit to see if we can improve this. 

For now, you could consider using the internal `Fulfil::Client#request` directly to make it do what you want. However, it's a private API, so it shouldn't be a long term approach.
